Robot2018/src/org/usfirst/frc/team199/Robot2018/commands/PIDTurn.java

```diff
@@ -18,19 +18,16 @@
 	DrivetrainInterface dt;
 	private PIDController turnController;
 
-	public PIDTurn(double targ, DrivetrainInterface dt, AHRS ahrs) {
+	public PIDTurn(double targ, DrivetrainInterface dt, PIDSource ahrs, SmartDashboardInterface sd) {
 		// Use requires() here to declare subsystem dependencies
 		// eg. requires(chassis);
 		target = targ;
 		this.dt = dt;
-		requires(Robot.dt);
-		// calculates the maximum turning speed in degrees/sec based on the max linear
-		// speed in inches/s and the distance (inches) between sides of the DT
-		double maxTurnSpeed = dt.getCurrentMaxSpeed() * 360
-				/ (Math.PI * Robot.getConst("Distance Between Wheels", 26.25));
-		double kf = 1 / (maxTurnSpeed * Robot.getConst("Default PID Update Time", 0.05));
-		turnController = new PIDController(Robot.getConst("TurnkP", 1), Robot.getConst("TurnkI", 0),
-				Robot.getConst("TurnkD", 0), kf, ahrs, this);
+		if (Robot.dt != null) {
+			requires(Robot.dt);
```

Comment:
what does this do? what the requires() method does is it stops the other commands that are using the same subsystem. https://wpilib.screenstepslive.com/s/currentCS/m/java/l/599745-scheduling-commands#WhatDoesThe&quot;requires&quot;MethodDo?

Reply:
This is also an addition from Corvin's tests.

Reply:
It's a check whether the code is running in a robot or as a unit test.  The Robot.dt is used as a proxy for an explicit flag for robot vs unit test.  Robot.dt is only initialized when robotInit() is called and is null when unit tests are run.  It would be better if there were a Robot.isUnitTesting() method that internally checked Robot.dt so that if the condition changes, then the checks in the rest of the code don't have to be rewritten.